rdd/RDDcounts.py

```
@@ -277,95 +322,229 @@ def create_RDD_counts_all_levels(self) -> pd.DataFrame:
             )
-        if reference_types is None:
-            filtered_df = self.counts[self.counts["level"] == level]
-            return filtered_df
-        filtered_df = self.counts[
-            (self.counts["reference_type"].isin(reference_types))
-            & (self.counts["level"] == level)
-        ]
+
+        if upper_level is not None and lower_level is not None:
+            if upper_level >= lower_level:
+                raise ValueError("upper_level must be lower than lower_level.")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)